services/navigation/path.go

```diff
@@ -0,0 +1,73 @@
+// Package navigation is the service that allows you to navigate along waypoints.
+package navigation
+
+import (
+	geo "github.com/kellydunn/golang-geo"
+	commonpb "go.viam.com/api/common/v1"
+	pb "go.viam.com/api/service/navigation/v1"
+)
+
+// Path describes a series of geo points the robot will travel through.
+type Path struct {
+	destinationWaypointID string
+	geoPoints             []*geo.Point
+}
+
+// NewPath constructs a Path from a slice of geo.Points and ID.
+func NewPath(id string, geoPoints []*geo.Point) *Path {
+	return &Path{
+		destinationWaypointID: id,
+		geoPoints:             geoPoints,
+	}
+}
+
+// DestinationWaypointID returns the ID of the Path.
+func (p *Path) DestinationWaypointID() string {
+	return p.destinationWaypointID
+}
+
+// GeoPoints returns the slice of geo.Points the Path is comprised of.
+func (p *Path) GeoPoints() []*geo.Point {
+	return p.geoPoints
+}
+
+// PathSliceToProto converts a slice of Path into an equivalent Protobuf message.
+func PathSliceToProto(paths []*Path) []*pb.Path {
```

Comment:
same question regarding whether it is possible for there to be nil ponters here. If its impossible then feel free to keep as is, but if nil pointers are possible then I think we need to check for them & return errors to the caller if they are present.